src/verbosity.jl

```diff
@@ -0,0 +1,85 @@
+function restore_callsite_source_position!(expr, src)
+    # used to fix the logging source file + line
+    # since we're lowering our verbose logging macros to the
+    # Logging.jl macros; otherwise, they would always report this (verbosity.jl)
+    # file as the logging callsite
+    expr.args[1].args[2] = src
+    return expr
+end
```

Comment:
I see what you are doing here.
Can we find a nicer way to do it?
The way i have seen Keno do this is strip all line numebrs from generate code and then just interpolate `__source__` as the first statement.
https://github.com/JuliaDiff/ChainRulesCore.jl/blob/2d75b4be102bb41ba3ac6df6dec8bb9617b20f0f/src/rule_definition_tools.jl#L178-L186

The way I have done it before is similar, but with creating the Expr directly so no line numbers endup there to begin with:
https://github.com/JuliaDiff/ChainRulesCore.jl/blob/2d75b4be102bb41ba3ac6df6dec8bb9617b20f0f/src/tangent_types/thunks.jl#L134

Reply:
Is there something problematic w/ this approach? I copied this from elsewhere; I can't remember if I wrote it or someone else.

Reply:
It's just very unclear what `expr.args[1].args[2]` is
I guess something that would make it ok would be something like
```julia
    @assert expr.head == :escape
    @assert expr.args[1].head == :macrocall
    @assert expr.args[1].args[2] isa LineNumberNode
    expr.args[1].args[2] = src
```
As a form of self-checking documentation.